sdktests/server_side_events_custom.go

```diff
@@ -18,66 +17,57 @@ func doServerSideCustomEventTests(t *ldtest.T) {
 
 	t.Run("data and metricValue parameters", doServerSideParameterizedCustomEventTests)
 
-	t.Run("index events are created for new users if not inlined", func(t *ldtest.T) {
```

Comment:
This is moved to `server_side_events_index.go`, which covers the `index` event behavior in detail. Now, in custom event tests where we expect an index event to be generated but it's not the focus of the test, we just use `IsIndexEvent()` to indicate very generally that the event payload will include one.